File: Lab12/src/basic-lambda.js
exports.handler = async (event) => {
    console.log("Incoming event:", JSON.stringify(event));
    const claims = event.requestContext?.authorizer?.claims || {};
    const groups = claims["cognito:groups"] || [];

    const path = event.resource;

    if (!groups.includes("admin")) {
        console.log("Access denied: User is not in the 'admins' group.");
        return {
            statusCode: 403,
            body: JSON.stringify({ error: "Access denied" })
        };
    }    
    const headers = event.headers || {};
    const token_header = headers['x-token-id']

    if (token_header !== undefined) {
        console.log(`Called endpoint with token header ${token_header}`)
    }

    const name = event.queryStringParameters?.name || "Unknown";

    const response = {
        message: `HELLO ${name.toUpperCase()} FROM NODE!`,
    };

    console.log("Response:", JSON.stringify(response));

    return {
        statusCode: 200,
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(response),
    };
}

File: Lab12/src/basic-lambda.py
import json
from datetime import datetime


def lambda_handler(event, context):
    print("Incoming event:", json.dumps(event))
    params = event.get("queryStringParameters") or {}
    name = params.get("name", "Unknown")
    headers = event.get("headers", {}) or {}
    token_header = headers.get("x-token-id") or None
    claims = event.get("requestContext", {}).get("authorizer", {}).get("claims", {})
    groups = claims.get("cognito:groups", [])
    if "student" not in groups and "admin" not in groups:
        print("Access denied: User is not in the 'student' or 'admin' group.")
        return {
            "statusCode": 403,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"error": "Access denied"})
        }
        
    if token_header:
        print(f"Called endpoint with token {token_header}")       
    # name = event.get("queryStringParameters", {}).get("name", "Unknown") # this is unsafe because queryStringParameters ends up being null or None if you don't pass any

    response = {
        "message": f"Hello {name} from Python!",
        "timestamp": datetime.utcnow().isoformat()
    }

    print("Response:", json.dumps(response))

    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps(response)
    }